subsys/bluetooth/host/conn.c

```diff
@@ -1353,6 +1456,57 @@ static void process_unack_tx(struct bt_conn *conn)
 	}
 }
 
+struct bt_conn *conn_lookup_handle(struct bt_conn *conns, size_t size,
+				   uint16_t handle)
+{
+	int i;
+
+	for (i = 0; i < size; i++) {
+		if (!atomic_get(&conns[i].ref)) {
+			continue;
+		}
+
+		/* We only care about connections with a valid handle */
+		if (!bt_conn_is_handle_valid(&conns[i])) {
+			continue;
+		}
+
+		if (conns[i].handle == handle) {
+			return bt_conn_ref(&conns[i]);
+		}
+	}
+
+	return NULL;
+}
+
+#if defined(CONFIG_BT_ISO)
+struct bt_conn *conn_lookup_iso(struct bt_conn *conn)
+{
+	int i;
+
+	for (i = 0; i < ARRAY_SIZE(iso_conns); i++) {
+		if (!atomic_get(&iso_conns[i].ref)) {
+			continue;
+		}
+
+		if (&iso_conns[i] == conn) {
+			return conn;
+		}
+
+		if (iso_conns[i].iso.acl == conn) {
+			return bt_conn_ref(&iso_conns[i]);
+		}
```

Comment:
This looks rather suspicious and seems like bad design: in some cases the function returns a new reference and in other cases it doesn't.

Reply:
Nice catch, I seems it doesn indeed expect a new ref in all cases, I will fix it.